src/makeCalculator.js

```diff
@@ -1,10 +1,39 @@
 'use strict';
 
-/**
- * @return {object}
- */
 function makeCalculator() {
-  // write code here
+  const resultObj = {
+    result: 0,
+
+    add(number) {
+      resultObj.result += number;
+    },
+
+    subtract(number) {
+      resultObj.result -= number;
+    },
+
+    multiply(number) {
+      resultObj.result *= number;
+    },
+
+    divide(number) {
+      resultObj.result /= number;
+    },
+
+    operate(callback, number) {
+      callback(number);
+
+      return this;
+    },
+
+    reset() {
+      resultObj.result = 0;
+
+      return this;
```

Comment:
[CODE KNOWLEDGE] - Similar to the `operate` method, use the method shorthand for the `reset` method. Replace `reset: function() {` with `reset() {` to comply with checklist item 3.